### ISSUE TITLE
Lack of loading screen / fetch 

### ISSUE THREAD

**Issue by danielsjoo** [keepid/keepid_client#507]
**Describe the bug**
When a worker enters the homepage, then they will have a homepage that that immediately says "No clients". If the request for clients from the server is not served quickly (or fails with an error), the worker could be confused and it will make the user experience more choppy.

**To Reproduce**
Log in as worker and go to home page

**Expected behavior**
While home page loads and is awaiting data from the server, there should a loading screen that looks passable. For instance, we can have these placeholding shimmer loading cards. https://codepen.io/muratbenli/pen/yLOYmgj 

<img width="1461" alt="Screenshot 2024-04-05 at 10 37 03 AM" src="https://github.com/keepid/keepid_client/assets/68679808/13db94db-845c-4fcd-a726-48c648ea4884">

**Additional Notes**
1. Should only require changes in WorkerLanding.tsx
2. This page should optimally be rendered only once on login and cached so the loading doesn't happen everytime the worker enters the homepage. That can be a separate issue, but if the tech stack ever changes (to Nextjs), that problem will be fixed by itself.
